uwblueprint/childrens-aid-society
# add providers state handling with add and delete functionality

## Notion ticket link

[Add Providers data handling](https://www.notion.so/uwblueprintexecs/134e1f159caa4f73939d11a700c325ad?v=39cf7d5a98384eb9af2bff2265d0ac8a&p=8de6a099d9484095b04f1450fb299ba7&pm=s)

Note: the ticket doesn't have much detail since I kind of impulse did this change, one thing just led to another lol. this is v bad practice 


## Implementation description
* Created ProviderDetails and Providers types
* Added Provider state management within AddChildPage and its relevant children components
* Modified PromptBox to take in name instead of firstName lastName
* Modified PromptBox to take in a delete function
* Implemented add and delete provider functionality through ChildProviderForm



## Steps to test
1. Go to /intake/add-child and the Providers section
2. Try adding providers with the Add Provider button
3. Try deleting providers with the trash icon
4. Try navigating around the Child Information pages to see if data persists locally



## What should reviewers focus on?
* how the Provider state/data is being handled and modified (it's quite messy rn and not great. I'm considering we start using Redux, but I'm seeing if we can manage just passing state around if our app is relatively simple)

## Checklist
